CSharp/Microsoft.Bot.Builder.Teams/Middlewares/TeamsTenantFilteringMiddleware.cs

```diff
@@ -51,23 +51,17 @@ public TeamsTenantFilteringMiddleware(IEnumerable<string> allowedTenantIds)
         /// </remarks>
         /// <seealso cref="ITurnContext" />
         /// <seealso cref="Schema.IActivity" />
-        public Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task OnTurnAsync(ITurnContext turnContext, NextDelegate next, CancellationToken cancellationToken = default(CancellationToken))
         {
-            // Ignoring cases where ChannelData is missing or does not contain TenantId.
-            if (turnContext.Activity.ChannelData != null)
-            {
-                TeamsChannelData teamsChannelData = turnContext.Activity.GetChannelData<TeamsChannelData>();
+            TeamsChannelData teamsChannelData = turnContext.Activity.GetChannelData<TeamsChannelData>();
```

Comment:
Did you check the behavior of this if channel data is missing entirely?

Reply:
Yes it returns null.